src/ows/src/main/java/org/geoserver/ows/XmlRequestReader.java

```diff
@@ -126,4 +130,73 @@ public int hashCode() {
     public String getServiceId() {
         return serviceId;
     }
+
+    /**
+     * Wrap request parsing failure in a SAX Document to prevent generic FileNotFound and
+     * ConnectionRefused messages.
+     *
+     * <p>This communicates that there is a problem with the XML Document while avoiding providing
+     * internal details.
+     *
+     * <p>The full stack trace is preserved to support development and debugging.
+     *
+     * @param t Parsing failure
+     * @return Clean SAXException for common parsing failures, or initial throwable
+     */
+    protected Exception cleanException(Exception t) {
+        if (t instanceof IOException) {
+            // FileNotFoundException, ConnectionException, etc...
+            // Log actual failure for debugging and troubleshooting
+            final Logger LOGGER = Logging.getLogger(this.getClass());
```

Comment:
(optional) I see LOGGER is only used here, but might make sense to put it in the class....

Reply:
I am actually looking up a specific logger from the subclass, so although it is unusual it is on purpose looking it up this way. Technically I could look it up in the constructor also.

Reply:
feedback addressed, created logger during constructor